x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -154,6 +157,16 @@ export class ActionExecutor {
           },
         };
 
+        for (const relatedSavedObject of relatedSavedObjects || []) {
+          event.kibana?.saved_objects?.push({
+            rel: SAVED_OBJECT_REL_PRIMARY,
```

Comment:
I agree, this is confusing for me too. There doesn't seem to be another option either

Reply:
Right, it is confusing.  We added this `rel: "primary"` field as a kind of primitive way of filtering out some of the searches across the event log by saved object.  Specifically, we didn't want someone with access to a connector, but not a rule, to see the "execute-action" event, which is logged when an action is queued to be executed by a rule.  Since that would "leak" the rule to someone who doesn't have access to it.

So this `rel` field comes into play when searching the event log by saved object.  We specifically require in the search that the saved object you're searching for should have a `rel: primary` field, otherwise we won't return that event log document.

This probably isn't even needed any more, since I believe that is the only case we DO NOT set the `rel` property - so I'll open an issue.

In this specific case, of the "action execute" event, it does seem like you'd want to find the event if you have either access to the connector or the rule, so it's set to "primary" on both.

And confusingly, if opposite of `rel: primary` is the absence of the `rel` field - not an actual value.  So there is only one valid string value of `rel`: `primary`.

Reply:
I was going to note that the README doc could stand to be updated, to help clarify this, but I see there is already some discussion in there, at the end of this section: https://github.com/elastic/kibana/tree/master/x-pack/plugins/event_log#event-documents

However, I notice that it's now out-of-date!  So I'll fix that up.  And take requests for additional cleanup / clarification of that section regarding the confusion aspect.

Reply:
Opened issue https://github.com/elastic/kibana/issues/102811 to track if we can get rid of `rel:primary`